servlet/src/main/scala/org/http4s/servlet/ServletIo.scala

```diff
@@ -189,6 +193,8 @@ final case class NonBlockingServletIo[F[_]: Async](chunkSize: Int) extends Servl
 
                   case Complete => cb(rightNone)
 
+                  case Canceled => cb(Left(new CancellationException("Servlet read was canceled")))
```

Comment:
I don't think this is ever going to be read, and we could just do a no-op cancel token (which is effectively how it has worked for years), but this at least gets us to a terminal state.  The same is true on the write side.